Increase stream read size and add MIME type fallback logic. .

herbertroth opened the pull request:
<!--

Before working on a contribution, you must determine on which branch you need to work:
- Bug fix: choose the latest maintenance branch `11.5`
- Feature/Improvement: choose `11.x` 

> All bug fixes merged into the latest maintenance branch are also merged to the current dev branch (`11.x`) on a regular basis.

## Please make sure your PR complies with all of the following points: 
- [ ] Read and accept our [contributing guidelines](/CONTRIBUTING.md) before you submit a PR.
- [ ] Features need to be proper documented in `doc/` 
- [ ] Bugfixes need a short guide how to reproduce them -> target branch is the oldest supported maintenance branch, e.g. `11.4` (see Readme.md for the list of supported versions)
- [ ] Meet all coding standards (see PhpStan actions) 

**Don't submit a PR if it doesn't comply, it'll be closed without a comment!**
-->  
  

## Changes in this pull request  
Resolves #

## Additional info

herbertroth committed:
Increase stream read size and add MIME type fallback logic.
.

herbertroth committed:
Fix condition to check array of extensions correctly

blankse commented on lib/Helper/MimeTypeHelper.php:
```diff
@@ -64,6 +64,39 @@ private function guessMimeTypeFromStream(mixed $stream): ?string
+            $mimeTypes = new MimeTypes();
+            $extensions = $mimeTypes->getExtensions($mimeType);
+
+            if (!empty($extensions)) {
```
Variable always exists
```suggestion
            if ($extensions) {
```

herbertroth committed:
Refactor MIME type guessing to improve metadata handling.

Replaced direct metadata access with a variable for clarity. Improved extension-based MIME type detection by extracting file extension from the URI and standardizing it to lowercase. Cleaned up conditions for better maintainability and readability.

herbertroth committed:
Simplify extension extraction logic in MimeTypeHelper

Combined null check and pathinfo extraction into a single line to streamline the code. This reduces redundancy and improves readability without altering functionality.

herbertroth committed:
Refactor MIME type resolution to use MimeTypes library

Replaced custom MIME type mapping with MimeTypes library for improved maintainability and broader support. This simplifies the code and reduces the need for manual updates to handle new file extensions.

herbertroth committed:
Fix variable assignment for mime type retrieval
